sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JacksonParser.scala

```diff
@@ -35,184 +34,307 @@ import org.apache.spark.util.Utils
 
 private[json] class SparkSQLJsonProcessingException(msg: String) extends RuntimeException(msg)
 
-object JacksonParser extends Logging {
+private[sql] class JacksonParser(schema: StructType, options: JSONOptions) extends Logging {
```

Comment:
Here, `JacksonParser` object was removed and `JacksonParser` class was added (to keep schema information).